src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -109,83 +110,106 @@ def project_root(self) -> Path:
         return self._project_root
 
     @property
-    def definition(self) -> Dict:
+    def definition(self) -> NativeAppSchema:
         return self._project_definition
 
     @cached_property
     def artifacts(self) -> List[ArtifactMapping]:
-        return [translate_artifact(item) for item in self.definition["artifacts"]]
+        return [translate_artifact(item) for item in self.definition.artifacts]
 
     @cached_property
     def deploy_root(self) -> Path:
-        return Path(self.project_root, self.definition["deploy_root"])
+        return Path(self.project_root, self.definition.deploy_root)
 
     @cached_property
     def package_scripts(self) -> List[str]:
         """
         Relative paths to package scripts from the project root.
         """
-        return self.definition.get("package", {}).get("scripts", [])
+        return (
+            self.definition.package.scripts
+            if (self.definition.package and self.definition.package.scripts)
+            else []
+        )
 
     @cached_property
     def stage_fqn(self) -> str:
-        return f'{self.package_name}.{self.definition["source_stage"]}'
+        return f"{self.package_name}.{self.definition.source_stage}"
 
     @cached_property
     def stage_schema(self) -> Optional[str]:
         return extract_schema(self.stage_fqn)
 
     @cached_property
     def package_warehouse(self) -> Optional[str]:
-        return self.definition.get("package", {}).get("warehouse", self._conn.warehouse)
+        return (
+            self.definition.package.warehouse
+            if (self.definition.package and self.definition.package.warehouse)
+            else self._conn.warehouse
+        )
```

Comment:
Personally I would remove all those ternary in favour of
```py
if self.definition.package and self.definition.package.warehouse:
    return self.definition.package.warehouse
return self._conn.warehouse
```
WDYT?

Reply:
Done